### PR TITLE
Backport to 11.0: Fixes a bug that prevents upgrades when there are no worker nodes

### PR DESCRIPTION
**Pull request by onderkalaci**
(cherry picked from commit bab4c0a8c363ce56acb1f03cba6988a9ed29455d)

Landing #6036 to release-11.0, adjust the commit so that we do not have any references to Citus 11, and the control file is adjusted properly

### CODE REVIEW

File: src/test/regress/expected/multi_extension.out [hunk 1]
```diff
@@ -1045,6 +1045,22 @@ SELECT * FROM multi_extension.print_extension_changes();
...
+
+-- Test downgrade script (result should be empty)
+ALTER EXTENSION citus UPDATE TO '11.0-2';
+SELECT * FROM multi_extension.print_extension_changes();
+ previous_object | current_object
+---------------------------------------------------------------------
+(0 rows)
+
+-- get back to the latest version
+ALTER EXTENSION citus UPDATE TO '11.0-3';
```
Comment (onderkalaci) [inline src/test/regress/expected/multi_extension.out:1063]: Hmm., why does:
```
-- show running version
SHOW citus.version;
 citus.version
---------------------------------------------------------------------
 11.0.2
(1 row)
```
in 5 lines later?

Comment (onderkalaci) [reply src/test/regress/expected/multi_extension.out:1063]: Do we change those when we prepare for the rerelease? https://github.com/citusdata/citus/commit/6663635593ed70a50f98f4688b9fb8ccfa0662d0

**Review by marcocitus — APPROVED** [review]

File: src/test/regress/sql/multi_extension.sql [hunk 2]
```diff
@@ -466,6 +466,16 @@ SELECT * FROM multi_extension.print_extension_changes();
...
+ALTER EXTENSION citus UPDATE TO '11.0-3';
+-- Snapshot of state at 11.0-3
+SELECT * FROM multi_extension.print_extension_changes();
+
+-- Test downgrade script (result should be empty)
+ALTER EXTENSION citus UPDATE TO '11.0-2';
+SELECT * FROM multi_extension.print_extension_changes();
+
+-- get back to the latest version
+ALTER EXTENSION citus UPDATE TO '11.0-3';
```
Comment (hanefi) [inline src/test/regress/sql/multi_extension.sql:478]:
```suggestion
```

We already got back to the latest version and printed changes.

**Review by hanefi — CHANGES_REQUESTED** [review]

File: src/test/regress/sql/multi_extension.sql [hunk 3]
```diff
@@ -466,6 +466,16 @@ SELECT * FROM multi_extension.print_extension_changes();
...
+
+-- Snapshot of state at 11.0-3
+ALTER EXTENSION citus UPDATE TO '11.0-3';
+SELECT * FROM multi_extension.print_extension_changes();
+
+-- Test downgrade script (result should be empty)
+ALTER EXTENSION citus UPDATE TO '11.0-2';
+ALTER EXTENSION citus UPDATE TO '11.0-3';
+SELECT * FROM multi_extension.print_extension_changes();
+
```
Comment (hanefi) [inline src/test/regress/sql/multi_extension.sql:478]:
```suggestion
```

File: src/test/regress/expected/multi_extension.out (hunk 1 above)
Comment (hanefi) [reply src/test/regress/expected/multi_extension.out]: Yes. We update that when we bump the versions in configure scripts.

Notice that this version is not the migration version. The formatting is slightly different.

**Review by hanefi — APPROVED** [review]
All looks good after we delete some lines from `multi_extension.out` as we recently removed one query from the sql file.

**Pull request merged**